### PR TITLE
[TTAHUB-2171] Table changes for merge goals

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Modify goals table and model to accommodate the incoming "merge goals" feature.

## How to test

Confirm changes make sense

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2171

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/models/goal.js [hunk 1]
```diff
@@ -93,6 +101,16 @@ export default (sequelize, DataTypes) => {
     previousStatus: {
       type: DataTypes.TEXT,
     },
+    mergedIntoId: {
```
Comment (GarrettEHill) [inline src/models/goal.js:104]: I know this is for merging goals, but in effect it directly correlates to the mapsTo convention we have been using recently. Personally I would prefer that naming convention.

Comment (thewatermethod) [reply src/models/goal.js:104]: makes sense, will change

File: src/models/goal.js [hunk 2]
```diff
@@ -43,6 +43,14 @@ export default (sequelize, DataTypes) => {
...
         as: 'resources',
       });
+      Goal.belongsTo(models.Goal, {
+        foreignKey: 'mapsTo',
+        as: 'parentGoal',
+      });
+      Goal.hasMany(models.Goal, {
+        foreignKey: 'mapsTo',
+        as: 'childGoals',
+      });
```
Comment (kryswisnaskas) [inline src/models/goal.js:53]: I've been thinking about the "`mapsTo`" convention, especially in the context of the parentGoal and childGoals associations. To align better with those, I'm leaning towards using `parentGoalId`. It ties in well with the hierarchy and still conveys the idea of an original or source goal in a merge context. What are your thoughts?

Comment (thewatermethod) [reply src/models/goal.js:53]: @GarrettEHill has one preference, you have another. 

I think the idea is communicated well enough with either "mapsTo" or "parentGoalId", so I'm not sure it matters to me.

Comment (kryswisnaskas) [reply src/models/goal.js:53]: @thewatermethod thanks for your flexibility on this. I appreciate that you are open to either convention.

@GarrettEHill, I get that you've introduced the "mapsTo convention and as the one close to the code, it's meaningful for you to have a convention that feels right. However, considering the potential move towards allowing users to use SQL queries and ensuring clarity both for as and future developers, I'm leaning towards a more descriptive and conventional name like parentGoalId.

I typically don't weigh in much on naming conventions, but this particular instance caught my attention due to its potential wider implications.

Can we discuss this and find a compromise?

Comment (thewatermethod) [reply src/models/goal.js:53]: I lean more towards descriptive names in general, but the convention being all over the place is persuasive.

"mapsToGoalId" or "mapsToParentGoalId"?

Comment (kryswisnaskas) [reply src/models/goal.js:53]: @thewatermethod, I appreciate the compromise. Given the context and future clarity, I would lean towards mapsToParentGoalId as it clearly describes the parent-child relationship. However, if we believe the parent context is assumed and want a slightly shorter name, mapsToGoalId works as well. Let's finalize on one of these.

Comment (thewatermethod) [reply src/models/goal.js:53]: Great, thanks @kryswisnaskas, I updated the name

**Review by kryswisnaskas — APPROVED** [review]
Looks good! 👍

**Pull request merged**